### PR TITLE
docs: Improve documents/tutorials and fix broken links

### PR DESCRIPTION
**Pull request by nicolehanjing**
**What this PR does / why we need it**:
1. fix the broken CPI installation tutorial link by referring to https://github.com/kubernetes/cloud-provider-vsphere/blob/6a93718c440343b65970f8a0bcdaaecff04a02b6/docs/book/tutorials/deploying_cloud_provider_vsphere_with_rbac.md; make updates according to the latest deployment process
2. tweak README file
3. fix broken links in the docs

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/kubernetes/cloud-provider-vsphere/issues/449

**Special notes for your reviewer**:

**Release note**:

```release-note
docs: Improve documents/tutorials and fix broken links
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,11 +15,11 @@ There is ongoing work for refactoring cloud providers out of the upstream reposi
 
 ## Quickstart
 
-Get started with Cloud controller manager for vSphere with Kubeadm with this [quickstart](https://cloud-provider-vsphere.sigs.k8s.io/tutorials/kubernetes-on-vsphere-with-kubeadm.html).
```
Comment (andrewsykim) [inline README.md:13]: Did https://cloud-provider-vsphere.sigs.k8s.io/tutorials/kubernetes-on-vsphere-with-kubeadm.html not work?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Kubernetes vSphere Cloud Provider
 
+![GitHub release (latest SemVer including pre-releases](https://img.shields.io/github/v/release/kubernetes/cloud-provider-vsphere?include_prereleases)
```
Comment (andrewsykim) [inline README.md:3]: Curious about this link to img.shields.io, what's it for?

Comment (nicolehanjing) [reply README.md:3]: it displays our latest release version of cpv

File: README.md (hunk 1 above)
Comment (nicolehanjing) [reply README.md:13]: the link is working, but I don't think it directs to the right place
I think we would want to visit the page that teaches deploying Cloud controller manager for vSphere with Kubeadm, right? 🤔

Comment (andrewsykim) [reply README.md:13]: I think https://github.com/kubernetes/cloud-provider-vsphere/blob/master/docs/book/tutorials/kubernetes-on-vsphere-with-kubeadm.md is the doc we want to share as a tutorial with kubeadm, and https://cloud-provider-vsphere.sigs.k8s.io/tutorials/kubernetes-on-vsphere-with-kubeadm.html is the docs site version that.

File: README.md [hunk 3]
```diff
@@ -14,7 +19,7 @@ Get started with Cloud controller manager for vSphere with Kubeadm with this [qu
 
 ## Quickstart using Helm
 
-Get started with Cloud controller manager for vSphere using Helm with this [Helm quickstart](https://cloud-provider-vsphere.sigs.k8s.io/tutorials/kubernetes-on-vsphere-with-helm.html).
```
Comment (andrewsykim) [inline README.md:17]: As a follow-up we should probably fix this link as well. Adding a link to tutorials here should do it https://github.com/kubernetes/cloud-provider-vsphere/blob/master/docs/book/SUMMARY.md#tutorials

Comment (andrewsykim) [reply README.md:17]: oh wait, I see you fixed it, then I think the old link can be used?

**Review by andrewsykim — COMMENTED** [review]
/approve

File: README.md (hunk 3 above)
Comment (nicolehanjing) [reply README.md:17]: yes we could keep using the old link

File: docs/book/cloud_provider_interface.md [hunk 4]
```diff
@@ -102,4 +102,35 @@ CPI is supported for vSphere versions 6.7U3 or later. CPI also requires a Kubern
 
 ## How do I install it?
 
-See the [Deploying cloud-provider-vsphere docs](https://cloud-provider-vsphere.sigs.k8s.io/tutorials/deploying_cloud_provider_vsphere_with_rbac.html) for install steps.
+### Deployment Overview
+
+Here are the steps to quickly deploying `cloud-provider-vpshere`:
```
Comment (lubronzhan) [inline docs/book/cloud_provider_interface.md:107]: vsphere typo

**Review by lubronzhan — APPROVED** [review]

**Review by lubronzhan — APPROVED** [review]

**Pull request merged**